### DISCUSSION TITLE
Add prefixes with vlan problem

### DISCUSSION THREAD

**Discussion by alamakot** [netbox-community/netbox#5494] [Q&A] [answered]
Hi,
I made own script to managed my netbox using API but I have one problem. 

I made new site then I added new IP address with mask and I created vlan group and vlans in this group. Finally I want to add prefixes with vlan but I can`t.
It is line from curl:
--data '{"prefix": "'$prefix'","vlan_group": {"name": "xxxx"},"vlan": {"vid": "'$vid'"},"status": "'${status,,}'" ,"site": {"name": "xxxx"}}'

Output: "Multiple objects match the provided attributes"

But if I add this from website:
Prefix Bulk import 
prefix,vlan_group,vlan,status,site
x.x.x.x/27,xxx,11,Active,xxx
It works. 

How can I add this by script using curl? 

Output indicates that it is impossible to identify the object because there are many vlanes with such an ID, but I thought that vlan group uniquely identifies them. Why it works from website?

Thanks.

**Comment by candlerb** [comment]
I am guessing you have multiple VLAN objects with vid (tag) 11.

You need to uniquely identify the VLAN to associate with the prefix - either using its database id (which is "id" not "vid"), or by name (VLANs don't appear to have a slug).

**Reply by alamakot** [reply]
You're right, I have many vid11 in different sites with the same names. 
This is some kind of solution to get their uniqeID from database, but as I mentioned  it works from website using only vid and vlan group. 
In one group I have only one vid11.

**Comment by candlerb** [comment, marked as answer]
You should be able to specify a tighter constraint to select the correct vlan, e.g.

```
... "vlan": {"vid": $vid, "group": {"name": "xxxx"}}, ...
```

You're trying to add a Prefix, right?  Prefix [doesn't have a vlan_group attribute](https://github.com/netbox-community/netbox/blob/develop/netbox/ipam/models.py#L361).  Hence vlan_group is an attribute of the vlan you're linking to, not the prefix itself.

**Reply by alamakot** [reply]
I changed data in curl on:
"vlan": {"vid": "11", "group": {"name": "xxx"}},
and it works.

Thank you very much.
